doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -981,106 +1003,98 @@ notations are given below. The optional :n:`@scope` is described in
 
 Global interpretation rules for notations
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+.. todo the evidence suggests that scopes are not organized as a stack,
+   so it's incorrect to refer to a stack of scopes below
+   1. One would expect the follow sequence would give an error because the last
+   command is not a pop.
+     Require Import ZArith.
+     Require Import PrimFloat.
+     Open Scope Z_scope.
+     Open Scope float_scope.
+     Close Scope Z_scope.
+   2. Print Visibility shows that items in the middle of the stack can be removed.
+
 At any time, the interpretation of a notation for a term is done within
-a *stack* of interpretation scopes and lonely notations. In case a
-notation has several interpretations, the actual interpretation is the
-one defined by (or in) the more recently declared (or opened) lonely
-notation (or interpretation scope) which defines this notation.
-Typically if a given notation is defined in some scope ``scope`` but has
-also an interpretation not assigned to a scope, then, if ``scope`` is open
-before the lonely interpretation is declared, then the lonely
-interpretation is used (and this is the case even if the
-interpretation of the notation in scope is given after the lonely
-interpretation: otherwise said, only the order of lonely
-interpretations and opening of scopes matters, and not the declaration
-of interpretations within a scope).
+a *stack* of notation scopes and lonely notations. If a
+notation is defined in multiple scopes, |Coq| uses the interpretation from
+the most recently opened notation scope or declared lonely notation.
+
+Use the :cmd:`Print Visibility` command to display the current notation scope stack.
 
 .. cmd:: Open Scope @scope
 
-   The command to add a scope to the interpretation scope stack is
-   :n:`Open Scope @scope`.
+   Adds a scope to the notation scope stack.
+
+   When the command appears in a section, the scope is only added within the section.
+   When the command is outside a section, the scope is exported as part of the current module.
+
+   This command supports the :attr:`local` and :attr:`global` attributes.
+   :attr:`local` adds the scope only in the current module.  :attr:`global`
+   adds the scope outside all currently open sections in the innermost open
+   module.  :attr:`global` has no effect if the command is not inside a section.
 
 .. cmd:: Close Scope @scope
 
-   It is also possible to remove a scope from the interpretation scope
-   stack by using the command :n:`Close Scope @scope`.
+   Removes a scope from the notation scope stack.
 
    Notice that this command does not only cancel the last :n:`Open Scope @scope`
    but all its invocations.
 
-.. note:: ``Open Scope`` and ``Close Scope`` do not survive the end of sections
-          where they occur. When defined outside of a section, they are exported
-          to the modules that import the module where they occur.
+   .. todo PR: invocations?
```

Comment:
All the previous times this command was called.

Reply:
Doing the same `Open Scope foo.` twice in a row is idempotent, right?  I guess opening foo, then bar, then foo is equivalent to opening bar and then foo.  A scope can only appear once in the stack, so the "invocations" is a bit misleading.

I will leave leave as a post-PR todo.

Reply:
sounds reasonable